src/vmm/src/lib.rs

```diff
@@ -601,6 +609,66 @@ impl Vmm {
         Ok(cpu_configs)
     }
 
+    /// Adds new vCPUs to VMM.
+    #[cfg(target_arch = "x86_64")]
+    pub fn hotplug_vcpus(
+        &mut self,
+        config: HotplugVcpuConfig,
+    ) -> Result<MachineConfigUpdate, HotplugVcpuError> {
+        use crate::logger::IncMetric;
+        #[allow(clippy::cast_possible_truncation)]
+        if config.vcpu_count < 1 {
+            return Err(HotplugVcpuError::VcpuCountTooLow);
+        } else if config.vcpu_count > MAX_SUPPORTED_VCPUS
+            || self.vcpus_handles.len() as u8 + config.vcpu_count > MAX_SUPPORTED_VCPUS
```

Comment:
`self.vcpus_handles.len() as u8 + config.vcpu_count > MAX_SUPPORTED_VCPUS` check should be enough here.

Reply:
No, that can overflow, see https://github.com/firecracker-microvm/firecracker/pull/4638#discussion_r1644272678

Reply:
Maybe in this case we just need to cast not to `u8` but to smth bigger?

Reply:
Or just do. `self.vcpus_handles.len().checked_add(config.vcpu_count)? > MAX_SUPPORTED_VCPUS`.